Use traffic_ctl in the trafficserver state and module.

jpeach opened the pull request:
### What does this PR do?

This is an RFC branch for updating the `trafficserver` state and module. It is completely untested; I'm pushing this early to solicit reviews and feedback.

Ping @jacksontj and @cedwards.
### What issues does this PR fix or reference?
#32752
### New Behavior

The `trafficserver` state and module will be compatible with upstream Traffic Server >= 7.0. 
### Tests written?

No. Tests are still TBD.

Traffic Server has deprecated traffic_line some time ago and subsumed
almost all of its functionality into traffic_ctl. This change checks whether
traffic_ctl is available and uses that in preference to traffic_line.

The only missing functionality from traffic_ctl is that of stopping
and starting the traffic_server process.

jpeach committed:
Use traffic_ctl in the trafficserver state and module.

Traffic Server has deprecated traffic_line some time ago and subsumed
almost all of its functionality into traffic_ctl. This change checks
whether traffic_ctl is available and uses that in preference to
traffic_line.

rallytime commented:
@jpeach This looks good to me. One thing that would be good to do here is to place a `.. versionadded:: Carbon` directive to each of the docstrings for the new functions you've added. Also, there are some small [lint errors](http://166.78.178.63:8080/job/PR/job/salt-pr-lint-n/1292/violations/file/salt/modules/trafficserver.py/) that need to be fixed before we can get this in.

cachedout commented:
Go Go Jenkins!

jpeach commented:
I added the last missing subcommands to ``traffic_ctl`` upstream, so this is now feature-equivalent. I also rebased and fixed the pylint errors.

I apologize, but I'm not sure when I'll have bandwidth to contribute tests for these :-/

cachedout commented:
Thank you very much, @jpeach
